dspace-api/src/main/java/org/dspace/storage/rdbms/DatabaseUtils.java

```diff
@@ -188,7 +188,9 @@ else if(argv[0].equalsIgnoreCase("migrate"))
                     // NOTE: This looks odd, but all we really need to do is ensure the
                     // DatabaseManager auto-initializes. It'll take care of the migration itself.
                     // Asking for our DB Name will ensure DatabaseManager.initialize() is called.
-                    DatabaseManager.getDbName();
+                    Connection connection = dataSource.getConnection();
+                    DatabaseUtils.updateDatabase(dataSource, connection);
+                    connection.close();
```

Comment:
You're right this will not work either. I will have to call initialize on the database manager before running the upgrade. I will also need to guard against upgrading twice. What I don't understand is how the code can work if you run migrate with arguments. Looks to me that it suffers from the same issue.